src/EnergyPlus/ResultsFramework.cc

```diff
@@ -387,9 +387,21 @@ namespace ResultsFramework {
             hourOfDay += 1;
         }
 
+        if (beginningOfInterval) {
+            if (hourOfDay == 24) {
+                hourOfDay = 0;
+            }
+            std::swap(hourOfDay, lastHour);
+            std::swap(curMin, lastMinute);
+        }
```

Comment:
Use the hour/minute from the previous call value if needed, resetting the hour to 0 if 24 is passed in.